src/lib.rs

```diff
@@ -140,6 +204,21 @@ impl SpinSleeper {
     /// **Windows**: Automatically selects the best native sleep accuracy generally achieving ~1ms
     /// native sleep accuracy, instead of default ~16ms.
     pub fn sleep(self, duration: Duration) {
+        /* likely bug here, at least for windozer.
+        The accuracy is not measured from point of call, it is a granularity measure.  In other
+        words, when you call sleep the accuracy is not "from the time you make that call" it is
+        rather until the next "tick" of the system clock.  So, if you call sleep(16'ms') and it is
+        half way between system ticks you will end up not waking up for 8 ms till next tick, which
+        is too early so it keeps sleeping till the next increment of 16 ms, meaning 24ms of actual
+        sleep time.
+
+        All said and done, you should use OS Sleep as much as possible even with all the problems.
+        Basically you take the desired sleep duration and chop it up by accuracy granularity, so
+        if the desired sleep is 30ms and accuracy is ~16ms, go ahead and sleep(16) then recheck the
+        timer, if >30ms, return obviously, if less "then" start using the spin loops to get closer
+        to the desired time.
+        */
+
```

Comment:
Sleep is only ever called once. So on Windows if you request to sleep 20ms, it asks to sleep 19ms then spins. If the OS sleeps more up to the accuracy we're fine, if it sleeps less we spin a little more.

This is the correct behaviour. The "accuracy" can be thought of as the worst case sleep overshoot, something this lib is trying to avoid.

Reply:
That's what I meant, I didn't see any 2x modifier in there so I think you possibly have a bug.  I.e. if the current timer has a frequency of say 16ms, calling sleep(15_ms) can sleep for 15ms up to 32ms, not what it seems like this is expecting of <16?  I might have read it wrong but it sure seemed like the expectation was based on when the call was made and not that it was somewhere inbetween the edge triggered timer ticks.

Reply:
I think the current method covers that to some extent. The accuracy (A) is the worst case overshoot, so if it does sleep an extra A over what we asked then we're fine. It's expecting `sleep(n)` to take at most `n + A`.

This accuracy A figure can also be configured, so you can bump it higher to spin a little more for accuracy.